src/vm/amd64/unixasmhelpers.S

```diff
@@ -172,6 +172,98 @@ NESTED_ENTRY JIT_RareDisableHelper, _TEXT, NoHandler
 
 NESTED_END JIT_RareDisableHelper, _TEXT
 
+#ifdef FEATURE_HIJACK
+
+//------------------------------------------------
+// OnHijackScalarTripThread
+//
+NESTED_ENTRY OnHijackScalarTripThread, _TEXT, NoHandler
```

Comment:
I have just realized that with returning structs in registers, we need to add saving restoring rdx as well here. That's the second return value register.